### PR TITLE
fix: pydantic ^2.9.0 needs 2 extra fields on to_argument

### PR DESCRIPTION
**Pull request by chrisemke**
## Description

when using to_pydantic with pydantic 2.9.0 it causes a mypy crash. Simmilar to what happend in this case: https://github.com/strawberry-graphql/strawberry/issues/3436

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #3620

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

## Summary by Sourcery

Fix a mypy crash by adding necessary fields to the to_pydantic method for compatibility with Pydantic 2.9.0.

Bug Fixes:
- Fix a mypy crash when using to_pydantic with Pydantic version 2.9.0 by adding two extra fields: 'model_strict' and 'is_root_model_root'.

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Krisque for the PR 👏

Get it here 👉 https://strawberry.rocks/release/(next)
```

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Krisque for the PR 👏

Get it here 👉 https://strawberry.rocks/release/(next)
```

**Review by patrick91 — APPROVED** [review]
Looks good to me! Thanks :)

Can you add a release file too?

**Comment by patrick91** [comment]
@chrisemke can you enable me to make commits here? I can try to fix the tests 😊

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Release type: patch
+
+Fix pydantic ^2.9.0 needs 2 extra fields on to_argument
+It was causing mypy crash
```
Comment (patrick91) [inline RELEASE.md:4]:
```suggestion
This releases adds support for Pydantic 2.9.0's Mypy plugin
```

**Review by patrick91 — APPROVED** [review]

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**